### PR TITLE
feat: GET /status/:cid to check pin & deal status

### LINKED ISSUE #78: Add status endpoint to check pin and deal info per cid
`GET /user/uploads/:cid`

Proposed respsonse shape:

```json
{
  "cid": "bafy",
  "dagSize": 101,
  "pins": [{
    "peerId": "12D3KooWR1Js",
    "peerName": "who?",
    "region": "where?",
    "status": "Pinned"
  }],
  "deals": [{
    "dealId": 12345,
    "miner": "f99",
    "status": "active",
    "activation": "<iso timestamp>",
    "pieceCid":  "baga",
    "dataCid":  "bafy",
    "dataModelSelector": "Links/0/Links"
  }]
}
```

This is the subset of properties in the current w3 schema that have some relevance to the user, flattened. Most `Batch` info is removed, apart from the `pieceCid`, the `dataCid` (the cid of the batch) the `dataModelSelector`, which, I'm told, could be used to extract the cid from the Piece.

The full shape of the schema that is available can be seen in the Fauna GraphQL explorer below for comparison
Screenshot 2021-07-13 at 10 16 57

### PR DESCRIPTION
**Pull request by olizilla**
- add **Unauthenticated** enpoint for checking pin and deal status by CID.
- adds mock and fixture for testing.
- only show pins and deals that are queued or active.

**Example reponse**

`GET /status/testcid`
```json
{
  "cid": "testcid",
  "dagSize": 101,
  "pins": [{
    "peerId": "12D3KooWR1Js",
    "peerName": "who?",
    "region": "where?",
    "status": "Pinned"
  }],
  "deals": [{
    "dealId": 12345,
    "miner": "f99",
    "status": "Active",
    "activation": "<iso timestamp>",
    "pieceCid": "baga",
    "dataCid": "bafy",
    "dataModelSelector": "Links/0/Links"
  }]
}
```

Fixes #78 

License: (Apache-2.0 AND MIT)
Signed-off-by: Oli Evans <oli@tableflip.io>

### CODE REVIEW

**Comment by olizilla** [comment]
@alanshaw do you have any magic CIDs that have pin and deal info in the staging db? this PR has only been tested with mock data. I'd like to see it deal with the real thing.

**Review by alanshaw — APPROVED** [review]
🚀 Awesome I love it!

File: packages/api/src/status.js [hunk 1]
```diff
@@ -13,29 +13,13 @@ const PIN_STATUS = new Set([
...
-    "pieceCid":  "baga",
-    "dataCid":  "bafy",
-    "dataModelSelector": "Links/0/Links"
-  }]
-}
-```
+/**
+ * Returns pin and deal status info for a given CID.
+ *
+ * @see {@link ../test/fixtures/status.json|Exmple resonse}
```
Comment (vasco-santos) [inline packages/api/src/status.js:19]: ❤️  was going to suggest this now

**Review by vasco-santos — APPROVED** [review]

File: packages/api/src/status.js (hunk 1 above)
Comment (vasco-santos) [reply packages/api/src/status.js:19]:
```suggestion
 * @see {@link ../test/fixtures/status.json|Example response}
```

**Comment by olizilla** [comment]
Thanks! I've added more tests for "we dont know that cid" and "no batch yet" and "batch but no deals"

**Review by alanshaw — APPROVED** [review]
Still a yes from me.

**Review by vasco-santos — APPROVED** [review]

**Pull request merged**